homeassistant/components/solarlog/sensor.py

```diff
@@ -47,39 +47,44 @@ async def async_setup_entry(hass, entry, async_add_entities):
 
     # Create a new sensor for each sensor type.
     entities = []
-    for sensor_key in SENSOR_TYPES:
-        sensor = SolarlogSensor(entry.entry_id, device_name, sensor_key, data)
+    for description in SENSOR_TYPES:
+        sensor = SolarlogSensor(entry.entry_id, device_name, description, data)
         entities.append(sensor)
-
     async_add_entities(entities, True)
     return True
 
 
 class SolarlogSensor(SensorEntity):
     """Representation of a Sensor."""
 
-    def __init__(self, entry_id, device_name, sensor_key, data):
+    def __init__(
+        self, entry_id, device_name, description: SolarLogSensorEntityDescription, data
+    ):
         """Initialize the sensor."""
+        self.entry_id = entry_id
         self.device_name = device_name
-        self.sensor_key = sensor_key
+        self.entity_description = description
         self.data = data
-        self.entry_id = entry_id
+
         self._state = None
 
-        self._json_key = SENSOR_TYPES[self.sensor_key][0]
-        self._label = SENSOR_TYPES[self.sensor_key][1]
-        self._unit_of_measurement = SENSOR_TYPES[self.sensor_key][2]
-        self._icon = SENSOR_TYPES[self.sensor_key][3]
+        self._key = self.entity_description.key
+        self._name = self.entity_description.name
+        self._icon = self.entity_description.icon
+        self._unit_of_measurement = self.entity_description.unit_of_measurement
+        self._state_class = self.entity_description.state_class
+        self._device_class = self.entity_description.device_class
+        self._json_key = self.entity_description.json_key
 
     @property
     def unique_id(self):
         """Return the unique id."""
-        return f"{self.entry_id}_{self.sensor_key}"
+        return f"{self.entry_id}_{self._key}"
 
     @property
     def name(self):
         """Return the name of the sensor."""
-        return f"{self.device_name} {self._label}"
+        return f"{self.device_name} {self._name}"
 
     @property
     def unit_of_measurement(self):
```

Comment:
with the entity description, you can remove this method.

Reply:
Removed